nautobot_golden_config/models.py

```diff
@@ -497,6 +501,19 @@ def __str__(self):
         return f"{self.device}"
 
 
+class GoldenConfigSettingManager(BaseManager.from_queryset(RestrictedQuerySet)):
+    """Manager for GoldenConfigSetting."""
+
+    def get_for_device(self, device):
+        """Return the highest weighted GoldenConfigSetting assigned to a device."""
+        if not isinstance(device, Device):
+            raise ValueError("The device argument must be a Device instance.")
+        dynamic_group = device.dynamic_groups.exclude(golden_config_setting__isnull=True)
```

Comment:
@jeffkala can you take a look at these queries?

Reply:
seems accurate

```
In [7]: device.dynamic_groups.exclude(golden_config_setting__isnull=True)
Out[7]: <DynamicGroupQuerySet [<DynamicGroup: GoldenConfigSetting Default Settings scope>, <DynamicGroup: gc2>]>

In [8]: dynamic_group.order_by("-golden_config_setting__weight").first().golden_config_setting
Out[8]: <GoldenConfigSetting: Golden Config Setting - Default Settings>

In [9]: device = Device.objects.first()

In [10]: device.dynamic_groups.exclude(golden_config_setting__isnull=True)
Out[10]: <DynamicGroupQuerySet [<DynamicGroup: GoldenConfigSetting Default Settings scope>, <DynamicGroup: gc2>]>

In [11]: dynamic_group.order_by("-golden_config_setting__weight").first().golden_config_setting
Out[11]: <GoldenConfigSetting: Golden Config Setting - gc2>
```